src/clout/core.cljc

```diff
@@ -0,0 +1,150 @@
+(ns clout.core
+  "A small language for routing."
+  (:require [clojure.string :as string]
+            [clojure.set :as set]
+            [instaparse.core :as insta]))
+
+(def ^:private re-chars (set "\\.*+|?()[]{}$^"))
+
+(defn- re-escape [s]
+  #?(:clj
+     (string/escape s #(if (re-chars %) (str \\ %)))
+     :cljs
+     (let [keys (seq re-chars) vals (->> keys (map #(str "\\" %)))
```

Comment:
Could this `let` clause be pulled outside of the function?

Reply:
I don't quite understand. What do you mean?